<script>
    function check(num){
        return new Promise((resolve,reject)=>{
            if(num<=10)
                resolve(num*num);
            else
                reject(num-5);
        });
    }
    var data = check(5).then((value)=>{
        return value*value;
    }).catch((error)=>{
        console.log("Catch : ",error);
        return error;
    })
    console.log(data);
    
    data.then((val)=>{
        console.log("second then : ",val);
    });

</script>